src/App.js

```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import './App.css';
+import AddProduct from "./components/AddProduct";
+import ProductList from "./components/ProductList";
+import {BrowserRouter, Route, Switch} from "react-router-dom";
+import ProductDetails from "./components/ProductDetails";
+import {connect} from "react-redux";
+
+function App() {
+    return (
+            <div className='wrapper'>
+                <AddProduct/>
+                <BrowserRouter>
+                    <Switch>
+                        <Route exact path='/'>
+                            <ProductList/>
+                        </Route>
+                        <Route exact path='/:id'>
+                            <ProductDetails/>
```

Comment:
Route components like ProductDetails, it's perfect place for code-splitting. Could you read this article and apply dynamic import for this component?
https://reactjs.org/docs/code-splitting.html